### ISSUE TITLE
BigtableSession should check for null credentials

### ISSUE THREAD

**Issue by joshfg** [GoogleCloudPlatform/cloud-bigtable-client#1478]
I am using the Bigtable emulator for development, which doesn't require authentication so is configured with `CredentialType.None`. The `CredentialFactory` returns `null` when asked for credentials of this type:
https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java#L138

`BigtableSession` uses this factory and does not check for `null` credentials, resulting in `null` being passed into `createNettyChannel`:
https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java#L470

This leads to failure with a `NullPointerException`:
```
Exception in thread "main" java.lang.NullPointerException: interceptor
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:787)
	at io.grpc.ClientInterceptors$InterceptorChannel.<init>(ClientInterceptors.java:113)
	at io.grpc.ClientInterceptors$InterceptorChannel.<init>(ClientInterceptors.java:107)
	at io.grpc.ClientInterceptors.intercept(ClientInterceptors.java:102)
	at io.grpc.internal.ManagedChannelImpl.<init>(ManagedChannelImpl.java:406)
	at io.grpc.internal.AbstractManagedChannelImplBuilder.build(AbstractManagedChannelImplBuilder.java:295)
	at com.google.cloud.bigtable.grpc.BigtableSession.createNettyChannel(BigtableSession.java:494)
	at com.google.cloud.bigtable.grpc.BigtableSession$5.create(BigtableSession.java:462)
	at com.google.cloud.bigtable.grpc.io.ChannelPool.<init>(ChannelPool.java:246)
	at com.google.cloud.bigtable.grpc.BigtableSession.createChannelPool(BigtableSession.java:458)
	at com.google.cloud.bigtable.grpc.BigtableSession.createChannelPool(BigtableSession.java:439)
	at com.google.cloud.bigtable.grpc.BigtableClusterUtilities.<init>(BigtableClusterUtilities.java:152)
	at com.google.cloud.bigtable.grpc.BigtableClusterUtilities.forAllInstances(BigtableClusterUtilities.java:69)
	at com.google.cloud.bigtable.grpc.BigtableClusterUtilities.lookupInstanceId(BigtableClusterUtilities.java:82)
	at com.google.cloud.bigtable.grpc.BigtableSession.resolveLegacyOptions(BigtableSession.java:278)
	at com.google.cloud.bigtable.grpc.BigtableSession.<init>(BigtableSession.java:205)
```

**Comment by igorbernstein2** [comment]
Hi joshfg,

Thanks for reporting the issue. Can you clarify which version of the client you are using? The line numbers from the stacktraces don't seem to line up with the current version of the client.
Can you try to reproduce the issue with the current version  (1.0.0-pre2)?

Thanks!

**Comment by joshfg** [comment]
Hi @igorbernstein2, I was using version 0.9.6.2 of bigtable-client-core. That version is being pulled in by a library I'm using but I'll try to replicate the issue with 1.0.0-pre2 tomorrow.

It's not very nice, but as a workaround, I'm able to use the Bigtable emulator by setting a fake credentials file / service account key.

**Comment by sduskis** [comment]
If you set the BIGTABLE_EMULATOR_HOST variable, things should just work with 0.9.6.2.  https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/bigtable-client-0.9.6.2/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java#L215

**Comment by mbrukman** [comment]
@joshfg – did you follow the instructions on [this page](https://cloud.google.com/bigtable/docs/emulator)? As @sduskis said, you need the `BIGTABLE_EMULATOR_HOST` env var, which can be set via:

```bash
> gcloud beta emulators bigtable start
> $(gcloud beta emulators bigtable env-init)
```

prior to the start of your application, and then it should just work without any code changes.

If you are getting that error in spite of following the instructions, then please let us know so that we can fix that. The goal is for emulator usage to be seamless and not require custom code.

**Comment by joshfg** [comment]
Hi @sduskis @mbrukman,
That error happened when I was setting the `BIGTABLE_EMULATOR_HOST` env var. I believe it's because of `CredentialOptions.nullOption()` being set here:
https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/bigtable-client-0.9.6.2/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java#L238

This leads to a null credentials interceptor being set here:
https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/bigtable-client-0.9.6.2/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java#L454

...because of the null returned here:
https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/CredentialInterceptorCache.java#L98

My workaround is to not set the `BIGTABLE_EMULATOR_HOST` env var, and instead configure the Bigtable options myself, setting a fake JSON credentials.

**Comment by igorbernstein2** [comment]
Hi @joshfg,

Can you confirm that this issue exists in 1.0.0-pre2?

Thanks!

**Comment by vertexclique** [comment]
Yes, this issue still exists and @joshfg 's comment is actually what I want to achieve.

**Comment by igorbernstein2** [comment]
@vertexclique is this issue present in the latest 1.3.0 release?

**Comment by sduskis** [comment]
We fixed this problem recently.  This ought to be fixed once we release 1.4.0
